### ISSUE TITLE
Clustering a Point Cloud

### ISSUE THREAD

**Issue by jlevy44** [rusty1s/pytorch_geometric#521]
If I wanted to cluster a point cloud (unsupervised point cloud segmentation), or pretrain graph level representations using a point cloud segmentation network, what would be the best approach?

Would I PointConv then XConv a few times with fps, then train the data using k-means loss? Or maybe some cross entropy loss function based on graclus assignments?

Also, if I load each point cloud via one point cloud per batch, would it be best to randomize the order of the points in cloud for each batch?

**Comment by rusty1s** [comment]
This is an interesting application, although I currently have no idea how to tackle this problem. I am not sure if `fps` is the best way to learn data-dependent clusters. Maybe you find [this paper](https://arxiv.org/abs/1905.12265) interesting as it is somehow related.

Randomizing the order of points is not really necessary since convolution and pooling operators are permutation-invariant (or like `fps`, randomize the input automatically).
